### PR TITLE
AWS S3: Add support for assume role duration seconds

### PR DESCRIPTION
**Pull request by rahull-p**
Changes Done 

Adds support for configuring session duration while assume role. Defaults to 3600 seconds 

Fixes https://github.com/Xuanwo/reqsign/issues/465

### CODE REVIEW

File: src/aws/config.rs [hunk 1]
```diff
@@ -79,6 +79,12 @@ pub struct Config {
     /// - profile config: `role_session_name`
     /// - default to `reqsign`.
     pub role_session_name: String,
+    /// `role_session_duration` value will be load from:
+    ///
+    /// - this field if it's `is_some`.
+    /// - profile config: `duration_seconds`
+    /// - default to `3600`.
+    pub role_session_duration: Option<i32>,
```
Comment (Xuanwo) [inline src/aws/config.rs:87]: I suggest keeping the same naming with AWS STS service to make it easier to understand.

**Review by Xuanwo — COMMENTED** [review]
Others look good to me

File: src/aws/config.rs (hunk 1 above)
Comment (rahull-p) [reply src/aws/config.rs]: Thanks for the review. Refactored the variable name for consistency.

File: src/aws/config.rs [hunk 2]
```diff
@@ -79,6 +79,12 @@ pub struct Config {
     /// - profile config: `role_session_name`
     /// - default to `reqsign`.
     pub role_session_name: String,
+    /// `duration_seconds` value will be load from:
+    ///
+    /// - this field if it's `is_some`.
+    /// - profile config: `duration_seconds`
+    /// - default to `3600`.
+    pub duration_seconds: Option<i32>,
```
Comment (Xuanwo) [inline src/aws/config.rs:87]: How about using `usize` instead? It's my personal preference in such cases.

Comment (rahull-p) [reply src/aws/config.rs]: Yeah that makes sense. Updated

**Review by Xuanwo — APPROVED** [review]
Thanks!

**Pull request merged**